File: Haskell/beginner/1866.hs
-- https://judge.beecrowd.com/pt/problems/view/1866
-- Conta

-- Main
main :: IO()
main = do

    usrCases <- readLn :: IO Int
    mapM_ (\_ -> do
        usrInput <- readLn :: IO Int
        if even usrInput
            then print 0
            else print 1
        ) [1..usrCases] -- Iterates between 1 and n

-- By Fernando Serra
-- https://github.com/fernandosserra

File: C/beginner/1866.c
// https://judge.beecrowd.com/pt/problems/view/1866
// Conta

// Libraries
#include <stdio.h>

// Main
int main(){

    int usrInput, usrCases, result;

    usrCases = 0;
    scanf("%d", &usrCases);

    for (int i = 0; i < usrCases; i++){
        scanf("%d", &usrInput);

        if (usrInput >= 1 && usrInput <= 100){
            if (usrInput % 2 == 0){
                result = 0;
            }
            else {
                result = 1;
            }
            printf("%d\n", result);
        }        
        else {
            printf("Invalid input.");
        }
    }

    return 0;    
}

// By: Fernando Serra
// https://github.com/fernandosserra